doggos/knowledge/clause.py

```diff
@@ -2,12 +2,146 @@
 from doggos.fuzzy_sets.fuzzy_set import FuzzySet
 from doggos.knowledge.linguistic_variable import LinguisticVariable
 
+import numpy as np
+from typing import NoReturn, Sequence
+
 
 class Clause:
+    """
+    Class representing a clause.
+    Clause is a pair of linguistic variable and fuzzy set:
+    https://en.wikipedia.org/wiki/Fuzzy_set
+    
+    Attributes
+    --------------------------------------------
+    __linguistic_variable : LinguisticVariable
+        linguistic variable, provides name and determines the domain of a fuzzy set
+    __gradiation_adjective : str
+        gradiation adjective, string representation of belonging level
+    __fuzzy_set : FuzzySet
+        fuzzy set provides its memebership function
+        
+    Methods
+    --------------------------------------------
+    get_value(self, x: float) -> Tuple[float, ...] or float
+        returns a value representing degree of belonging to a fuzzy set
+        
+    Examples
+    --------------------------------------------
+    >>> domain = Domain(0,10,0.01)
+    >>> ling_var = LinguisticVariable('Temperature', domain)
+    >>> f_set = T1FuzzySet(lambda x: 0 if x < 0 else 1)
+    >>> clause = Clause(ling_var, 'Medium', f_set)
+    >>> clause.get_value(1.2)
+    1.0
+    
+    """
 
-    __lingustic_variable: LinguisticVariable
-    __gradiation_adjectice: str
+    __linguistic_variable: LinguisticVariable
+    __gradiation_adjective: str
     __fuzzy_set: FuzzySet
+    
+    def __init__(self, linguistic_variable: LinguisticVariable, gradiation_adjective: str, fuzzy_set: FuzzySet):
+        """
+        Creates clause with given linguistic variable, gradiation adjective and fuzzy set.
+
+        :param linguistic_variable: linguistic variable, provides name and determines the domain of a fuzzy set
+        :param gradiation_adjective: gradiation adjective, string representation of belonging level
+        :param fuzzy_set: fuzzy set provides its memebership function
+        """
+        if not isinstance(linguistic_variable, LinguisticVariable):
+            raise TypeError("Linguistic variable must be LinguisticVariable type")
+        if not isinstance(fuzzy_set, FuzzySet):
+            raise TypeError("Fuzzy set must be FuzzySet type")
+        
+        self.__linguistic_variable = linguistic_variable
+        self.__gradiation_adjective = gradiation_adjective
+        self.__fuzzy_set = fuzzy_set
+        self.__values = self._calculate_values()
 
     def get_value(self, x: float) -> MembershipDegree:
-        pass
+        """
+        returns a value representing degree of belonging to a fuzzy set
+        :param x: degree of belonging
+        """
+        return self.__values[self._find_index(x)]
+    
+    def _calculate_values(self) -> Sequence[MembershipDegree]:
+        """
+        Calculates values for every element in domain
+        :return: arrau of membership degrees
+        """
+        return self.fuzzy_set(self.linguistic_variable.domain())
```

Comment:
You can change fuzzy_set to self.__fuzzy_set. That way it will be easier to distinguish between field of clause with that name and setter method.